src/System.Private.CoreLib/shared/System/AggregateException.cs

```diff
@@ -438,16 +438,12 @@ public override string ToString()
             StringBuilder text = new StringBuilder();
             text.Append(base.ToString());
 
-            for (int i = 0; i < m_innerExceptions.Count; i++)
+            foreach (Exception ex in m_innerExceptions)
             {
-                if (m_innerExceptions[i] == InnerException)
+                if (object.ReferenceEquals(ex, InnerException))
```

Comment:
Why is == inappropriate here?

Reply:
We want to filter out the exact same object - figured it was clearer. Also a subclass could have overridden the equals operator.

Reply:
> Also a subclass could have overridden the equals operator.

That's actually not how `==` works.  Unless the strongly-typed variable being used defines an `operator ==`, this will be emitted with the exact same behavior as ReferenceEquals.  It doesn't use `Equals`. e.g. this will output `false`:
```C#
using System;

class Program
{
    static void Main() => Console.WriteLine(new Program() == new Program());

    public override bool Equals(object obj) => true;
}
```

Reply:
I had forgotten this. In the case where this is known to be true (as here), do we prefer to use `==` for brevity, or O.RE to be explicit about intent? I assume the former as we don't seem to use O.RE that much.

Reply:
It often ends up being personal preference.

Reply:
Roslyn [compiles ReferenceEquals into the same IL](https://github.com/dotnet/roslyn/blob/34842271875f96a5a1bbebdf6c7d3f6f703cc708/src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs#L205
) as what you from `==`, so it is just a style difference. Similar to `== null` vs. `is null`.